tutorials/llm/llama-3/pruning-distillation/introduction.ipynb

```diff
@@ -0,0 +1,190 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eaef2c7d-41f7-41ad-a76a-2d714e9c35de",
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Split into train, test and val files\n",
+    "\n",
+    "import json\n",
+    "import os\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "# Load the WikiText-103 dataset\n",
+    "dataset = load_dataset(\"wikitext\", \"wikitext-103-v1\")\n",
+    "\n",
+    "# Define the destination folder\n",
+    "data_folder = 'wikitext-data'\n",
+    "os.makedirs(data_folder, exist_ok=True)\n",
+    "\n",
+    "# Define file paths and destination paths\n",
+    "file_paths = {\n",
+    "    'train': os.path.join(data_folder, 'wikitext-train.jsonl'),\n",
+    "    'validation': os.path.join(data_folder, 'wikitext-val.jsonl'),\n",
+    "    'test': os.path.join(data_folder, 'wikitext-test.jsonl')\n",
+    "}\n",
+    "\n",
+    "# Function to save dataset split to a JSONL file\n",
+    "def save_to_jsonl(file_path, data):\n",
+    "    with open(file_path, 'w') as file:\n",
+    "        for item in data:\n",
+    "            file.write(json.dumps(item) + '\\n')\n",
+    "\n",
+    "# Define splits\n",
+    "splits = [\"train\", \"validation\", \"test\"]\n",
+    "\n",
+    "# Save splits to JSONL files and calculate their sizes\n",
+    "for split in splits:\n",
+    "    if split in dataset:\n",
+    "        save_to_jsonl(file_paths[split], dataset[split])\n",
+    "    else:\n",
+    "        print(f\"Split {split} not found in the dataset.\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2d0cc359-0598-40aa-af80-9503ecd4dac1",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "---\n",
+    "##  Step-by-step instructions\n",
+    "\n",
+    "This workflow is structured into seven notebooks:\n",
+    "1. [Prepare the dataset](./01_data_preparation.ipynb)\n",
+    "2. [Finetune the teacher on the dataset](./02_teacher_finetuning.ipynb)\n",
+    "3. Prune the finetuned-teacher model to create a student \n",
+    "   - 3.a. [Using depth-pruning](./03_a_depth_pruning.ipynb)\n",
+    "   - 3.b. [Using width-pruning](./03_b_width_pruning.ipynb)\n",
+    "4. Distill knowledge from teacher into student\n",
+    "   - 4.a. [Using depth-pruned student](./04_a_distilling_depth_pruned_student.ipynb)\n",
+    "   - 4.b. [Using width-pruned student](./04_b_distilling_width_pruned_student.ipynb)\n",
+    "5. [Display the validation loss](./05_display_results.ipynb)\n",
+    "\n",
+    "> `NOTE:` We are exploring two methods to prune the finetuned teacher model: [depth-pruning](./03_a_depth_pruning.ipynb) and [width-pruning](./03_b_width_pruning.ipynb). Per the [tech report](https://arxiv.org/pdf/2408.11796), we can observe that width-pruning generally outperforms depth-pruning so users can choose to perform either [depth-pruning](./03_a_depth_pruning.ipynb) or [width-pruning](./03_b_width_pruning.ipynb) or both methods."
```

Comment:
fix punctuation

    "> `NOTE:` We are exploring two methods to prune the fine-tuned teacher model: [depth-pruning](./03_a_depth_pruning.ipynb) and [width-pruning](./03_b_width_pruning.ipynb). Per the [tech report](https://arxiv.org/pdf/2408.11796), we can observe that width-pruning generally outperforms depth-pruning so users can choose to perform either [depth-pruning](./03_a_depth_pruning.ipynb) or [width-pruning](./03_b_width_pruning.ipynb) or both methods."